integtest/spec/all_books_change_detection_spec.rb

```diff
@@ -595,26 +676,45 @@ def chapter(index)
         end
         context 'because there is an unrelated change in a mapped branch' do
           build_one_book_out_of_two_repos_twice(
-            before_first_build: lambda do |src|
+            before_first_build: lambda do |src, _config|
               book = src.book 'Test'
               repo2 = src.repo 'repo2'
               book.source repo2, 'included.asciidoc',
                           map_branches: { 'master': 'override' }
               repo2.switch_to_new_branch 'override'
             end,
-            before_second_build: lambda do |src|
+            before_second_build: lambda do |src, _config|
               repo2 = src.repo 'repo2'
               repo2.write 'garbage', 'junk'
               repo2.commit 'adding junk'
             end
           )
           include_examples 'second build is noop'
         end
+        context 'because we add a new branch and specify --keep_hash ' \
+                'and --sub_dir on the new branch with one repo' do
```

Comment:
This reproduced the problem we saw on Thursday with the elasticsearch and kibana repos. The book was built from more than one source, one of which was `--sub_dir`ed.